app/controllers/external_controller.rb

```diff
@@ -67,12 +67,12 @@ def create_user
     # TODO: - Ahmad: deal with errors
     redirect_location = cookies[:location]
     cookies.delete(:location)
-    return redirect_to redirect_location if redirect_location&.match?('\A\/rooms\/\w{3}-\w{3}-\w{3}-\w{3}\/join\z')
+    return redirect_to redirect_location if redirect_location&.match?(%r{\A(/\w{1,32}){0,3}/rooms/\w{3}-\w{3}-\w{3}-\w{3}/join\z})
```

Comment:
This matching seems a bit weird to me - why not use the "#{root_path}" macro here?
This will only match root_paths with up to three subdirs and each being from one to 32 characters long. An arbitrary limitation which probably won't bite anyone, but why risk it?

Reply:
I see what you mean, I'll make sure to keep this in mind as we work on this issue.